RecoLocalTracker/SiPixelRecHits/plugins/pixelProducerHelpers.h

```diff
@@ -0,0 +1,94 @@
+#ifndef pixelProducerHelpers_H
+#define pixelProducerHelpers_H
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+
+// Geometry
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+
+#include "CUDADataFormats/SiPixelCluster/interface/gpuClusteringConstants.h"
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DHeterogeneous.h"
+#include "CUDADataFormats/Common/interface/HostProduct.h"
+using HMSstorage = HostProduct<unsigned int[]>;
+
+// make    code rules happy
+namespace pixelProducerHelpers {
+
+  inline TrackingRecHit2DSOAView const* storeSoA(edm::Event& iEvent,
+                                                 std::unique_ptr<TrackingRecHit2DHost>& input,
```

Comment:
In case these get worked on, would it work if `input` would be moved-in instead of passing by reference? I.e.
```suggestion
                                                 std::unique_ptr<TrackingRecHit2DHost> input,
```